ui/packages/catalog-ui-search/src/main/webapp/react-component/container/result-sort/result-sort.tsx

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser General Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+
+import { hot } from 'react-hot-loader'
+import * as React from 'react'
+import ResultSortPresentation from '../../presentation/result-sort'
+
+const Backbone = require('backbone')
+const user = require('../../../component/singletons/user-instance.js')
+
+type Props = {
+  closeDropdown: () => void
+}
+
+type State = {
+  collection: Backbone.Collection<Backbone.Model>
+}
+
+export default hot(module)(
+  class ResultSort extends React.Component<Props, State> {
+    constructor(props: Props) {
+      super(props)
+      const resultSort = user
+        .get('user')
+        .get('preferences')
+        .get('resultSort')
+
+      this.state = {
+        collection: new Backbone.Collection(resultSort),
```

Comment:
Putting a backbone collection in to your state doesn't guarantee react will re-render on a collection / model change.

Reply:
For that, you need something like [withListenTo](https://github.com/codice/ddf/blob/master/ui/packages/catalog-ui-search/src/main/webapp/react-component/container/backbone-container/backbone-container.tsx#L27).

Reply:
This component isn't rendering based on the collection's state; that is done in the SortCollectionView, so I don't believe anything needs to be listened to here.  https://github.com/codice/ddf/pull/4337/files#diff-b958343a63c81835eb0e7dfdd2ae3987R53